Fixed multiple settings menus

drewvolz opened the pull request:
This prevents the settings button from being mashed quickly which results in pushing two or more views to the stack. This closes #155.

drewvolz committed:
Fixed multiple settings menus from being summoned on the home screen

hawkrives commented on app.js:
```diff
@@ -135,19 +135,27 @@ const styles = StyleSheet.create({
-            sceneConfig: Navigator.SceneConfigs.FloatFromBottom,
-          })}
+          onPress={settingsButtonActive ? () => {} : () => {
+            settingsButtonActive = true
```
Then call that callback here

hawkrives replied:
Ignore this comment

hawkrives commented on app.js:
```diff
@@ -135,19 +135,27 @@ const styles = StyleSheet.create({
-            index: route.index + 1,
-            sceneConfig: Navigator.SceneConfigs.FloatFromBottom,
-          })}
+          onPress={settingsButtonActive ? () => {} : () => {
```
You might also name the noop function, since we're naming the other.

hawkrives replied:
Ignore this comment

hawkrives commented on app.js:
```diff
@@ -157,7 +165,10 @@ function LeftButton(route, navigator, index, navState) {
-          onPress={() => navigator.pop()}
+          onPress={() => {
+            navigator.pop()
+            settingsButtonActive = false
```
This online function should probably be extracted to a member function,`closeSettingsScreen` or something.

hawkrives replied:
This is still a comment I'd like to have addressed.

hawkrives commented on app.js:
```diff
@@ -135,19 +135,27 @@ const styles = StyleSheet.create({
-            index: route.index + 1,
-            sceneConfig: Navigator.SceneConfigs.FloatFromBottom,
-          })}
+          onPress={settingsButtonActive ? () => {} : () => {
```
Import `noop` from lodash and use it here, and move the `settingsButtonActive = true` line into the `openSettings` function.

`onPress={settingsButtonActive ? noop : () => openSettings(route, navigator)}`

hawkrives reviewed (changes requested):
Moar feedbacks

drewvolz committed:
Updated changes according to review

- Extract logic into an openSettings function
- Extract logic into a closeSettings function
- Import noop from lodash and use it

hawkrives commented on app.js:
```diff
@@ -146,16 +148,18 @@ function openSettings(route, navigator) {
 
+function closeSettings() {
+  navigator.pop()
+  settingsButtonActive = false
```
Flip these?

hawkrives replied:
It also needs to accept `navigator` as a prop

drewvolz replied:
Done.

hawkrives commented on app.js:
```diff
@@ -146,16 +148,18 @@ function openSettings(route, navigator) {
-            settingsButtonActive = true
-            openSettings(route, navigator)
-          }}
+          onPress={settingsButtonActive ? noop : () => openSettings(route, navigator)}
```
✅

hawkrives commented on app.js:
```diff
@@ -166,8 +170,7 @@ function LeftButton(route, navigator, index, navState) {
-            navigator.pop()
-            settingsButtonActive = false
+            closeSettings()
           }}
```
Almost

We need to give `navigator` to the function

`onPress={() => closeSettings(navigator)}`

drewvolz committed:
Update according to CI failure

Forgot to pass in a variable after extracting the logic

hawkrives reviewed (changes requested):
Almost there

hawkrives commented on app.js:
```diff
@@ -157,7 +169,9 @@ function LeftButton(route, navigator, index, navState) {
         <TouchableOpacity
           style={[styles.backButton, {marginLeft: 10, marginVertical: Platform.OS === 'android' ? 21 : 16}]}
-          onPress={() => navigator.pop()}
+          onPress={() => {
```
put this on one line and remove the outer set of braces?

drewvolz replied:
Sure

hawkrives commented on app.js:
```diff
@@ -135,19 +136,30 @@ const styles = StyleSheet.create({
+
+function closeSettings(navigator) {
+  navigator.pop()
+  settingsButtonActive = false
```
and should we flip these two?

I don't think there'll be any functional difference

drewvolz replied:
We could.

drewvolz committed:
More changes according to review

- Flip order of pop and bool
- Make a multiple-liner a one-liner
